app/src/lib/branch.ts

```diff
@@ -41,3 +44,29 @@ export function findDefaultUpstreamBranch(
 
   return foundBranch !== undefined ? foundBranch : null
 }
+
+/**
+ * 
+ * @param repository The repository to use.
+ * @param branchesState The branches state of the repository.
+ * @returns The default branch of the user's contribution target, or null if it's not known.
+ * 
+ * This method will return the fork's upstream default branch, if the user
+ * is contributing to the parent repository.
+ * 
+ * Otherwise, this method will return the default branch of the passed in repository.
+ */
+export function findContributionTargetDefaultBranch(
+  repository: Repository,
+  branchesState: IBranchesState
+): Branch | null {
+  let contributionTargetDefaultBranch: Branch | null
+    if (isRepositoryWithGitHubRepository(repository)) {
+      contributionTargetDefaultBranch = findDefaultUpstreamBranch(
+        repository,
+        branchesState.allBranches
+      )
+    }
+    contributionTargetDefaultBranch ??= branchesState.defaultBranch
+    return contributionTargetDefaultBranch
+}
```

Comment:
> Seems like the build is failing due to an outdated prettier dependency not knowing how to interpret the syntax. Updating this dependency causes ALOT of other files to be modified/linted, so I guess we'll have to wait for https://github.com/desktop/desktop/pull/14070 to be merged.

Yeah, we're looking at that in #14205. In the meantime though I think you can pretty easily avoid the `??=` operator here. We typically prefer avoiding temporary mutable variables like that anyway (see https://github.com/desktop/desktop/blob/24d89a800755705c0e9c3f1239d3b3ac224721f5/docs/contributing/engineering-values.md#immutability-is-good)

```suggestion
export function findContributionTargetDefaultBranch(
  repository: Repository,
  { allBranches, defaultBranch }: IBranchesState
): Branch | null {
  return isRepositoryWithGitHubRepository(repository)
    ? findDefaultUpstreamBranch(repository, allBranches) ?? defaultBranch
    : defaultBranch
}
```

Reply:
You can destructure objects *in the function declaration itself*??? 😮